### ISSUE TITLE
The Effects of Regularization and Data Augmentation are Class Dependent

### ISSUE THREAD

**Issue by rogergranada** [unico-labs/ml-reading-club#86]
Regularization is a fundamental technique to prevent over-fitting and to improve generalization performances by constraining a model’s complexity. Current Deep Networks heavily rely on regularizers such as Data-Augmentation (DA) or weight-decay, and employ structural risk minimization, i.e. cross-validation, to select the optimal regularization hyper-parameters. In this study, we demonstrate that techniques such as DA or weight decay produce a model with a reduced complexity that is unfair across classes. The optimal amount of DA or weight decay found from cross-validation leads to disastrous model performances on some classes e.g. on Imagenet with a resnet50, the “barn spider” classification test accuracy falls from 68% to 46% only by introducing random crop DA during training. Even more surprising, such performance drop also appears when introducing uninformative regularization techniques such as weight decay. Those results demonstrate that our search for ever increasing generalization performance -averaged over all classes and samples- has left us with models and regularizers that silently sacrifice performances on some classes. This scenario can become dangerous when deploying a model on downstream tasks e.g. an Imagenet pre-trained resnet50 deployed on INaturalist sees its performances fall from 70% to 30% on class #8889 when introducing random crop DA during the Imagenet pre-training phase. Those results demonstrate that designing novel regularizers without class-dependent bias remains an open research question.

**Paper**: https://arxiv.org/pdf/2204.03632.pdf

**Comment by mchelem** [comment]
THIS

**Comment by rogergranada** [comment]
THIS

**Comment by rafascm** [comment]
THIS

**Comment by mchelem** [comment]
THIS

**Comment by gustavofuhr** [comment]
THIS

**Comment by amandaleonel** [comment]
THIS
